solutions/instances/variables.tf

```diff
@@ -339,6 +267,36 @@ variable "cos_instance_tags" {
   default     = []
 }
 
+variable "enable_archive_bucket_retention" {
+  description = "Whether retention is enabled for the Object Storage bucket."
+  type        = bool
+  default     = false
+}
```

Comment:
I'm not sure we should add this. logdna is no longer supported in our DA. So users will not be adding new log archive buckets. We allow users to keep managing existing archive buckets using the `manage_log_archive_cos_bucket` but they won't be creating new ones, and its not possible to enable retention on an existing bucket that already has objects in it.